### PR TITLE
Re-focus resident key definitions around allowCredentials aspect

### LINKED ISSUE #1197: Re-focus resident key descriptions on allowCredentials aspect
@agl remarked on the 2019-04-03 WG call that the terminology for [resident keys](https://www.w3.org/TR/webauthn/#resident-credential) (aka "resident credentials") conflates two disctinct aspects:

1. A resident key is stored in the authenticator.
2. A resident key can be used with empty `allowCredentials`, e.g.:
```
var options = {
  challenge: new Uint8Array([4,101,15 /* 29 more random bytes generated 
                                        by the server */]),
  timeout: 60000,  // 1 minute
  // we're letting allowCredentials default to an empty sequence
};
navigator.credentials.get({ "publicKey": options })
```
...and that of the two, (2) is usually the more interesting one from an RP perspective. There was broad agreement on the call that we should reformulate the terminology to focus more on aspect (2) than (1).

See also: https://github.com/w3c/webauthn/issues/991#issuecomment-479645567

### PR DESCRIPTION
**Pull request by emlun**
Closes #1197.

***
<a href="https://pr-preview.s3.amazonaws.com/w3c/webauthn/pull/1249.html" title="Last updated on Jul 3, 2019, 2:47 PM UTC (4154a53)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/webauthn/1249/126223e...4154a53.html" title="Last updated on Jul 3, 2019, 2:47 PM UTC (4154a53)">Diff</a>

### CODE REVIEW

File: index.bs [hunk 1]
```diff
@@ -918,14 +918,19 @@ The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "S
...
-:: A [=Client-side-resident Public Key Credential Source=], or [=Resident Credential=] for short, is a [=public key credential
-    source=] whose [=credential private key=] is stored in the [=authenticator=], [=client=] or [=client device=]. Such
-    [=client-side=] storage requires a [=resident credential capable=] [=authenticator=] and has the property that the [=authenticator=]
-    is able to select the [=credential private key=] given
-    only an [=RP ID=], possibly with user assistance (e.g., by providing the user a pick list of [=public key credential|credentials=] [=scoped=] to the [=RP ID=]).
-    By definition, the [=credential private key=] is always exclusively controlled by the [=authenticator=]. In the case of a
-    [=resident credential=], the [=authenticator=] might offload storage of wrapped key material to the
-    [=client device=], but the [=client device=] is not expected to offload the key storage to remote entities (e.g., [=[WRP]=] Server).
+:: A [=Client-side-resident Public Key Credential Source=], or [=Resident Credential=] for short,
+    is a [=public key credential source=] that can be used in an [=authentication ceremony=]
```
Comment (equalsJeffH) [inline index.bs]:
```suggestion
    is a [=public key credential source=] that is discoverable and usable in  [=authentication ceremonies=]
```

File: index.bs [hunk 2]
```diff
@@ -918,14 +918,19 @@ The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "S
...
+    without providing the [=authenticator=] with a [=credential ID=], i.e.,
```
Comment (equalsJeffH) [inline index.bs]:
```suggestion
    without providing [=credential ID=]s, i.e.,
```

File: index.bs [hunk 3]
```diff
@@ -918,14 +918,19 @@ The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "S
...
+    in an [=authentication ceremony=] with an [=list/empty=] {{PublicKeyCredentialRequestOptions/allowCredentials}} argument.
```
Comment (equalsJeffH) [inline index.bs]:
```suggestion
     calling {{CredentialsContainer/get()|navigator.credentials.get()}} with an [=list/empty=] {{PublicKeyCredentialRequestOptions/allowCredentials}} argument.
```

File: index.bs [hunk 4]
```diff
@@ -918,14 +918,19 @@ The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "S
...
+
+    As a consequence, a [=resident credential capable=] [=authenticator=] can generate an [=assertion signature=]
+    for a [=resident credential=] given only an [=RP ID=],
+    which in turn means that the corresponding [=credential private key=]
```
Comment (equalsJeffH) [inline index.bs]: should "[=credential private key=]" here be "[=public key credential source=]" ?

File: index.bs [hunk 5]
```diff
@@ -918,14 +918,19 @@ The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "S
...
+    is stored in the [=authenticator=], [=client=] or [=client device=].
```
Comment (equalsJeffH) [inline index.bs]: should we use "or [=client platform=]" here rather than ", [=client=] or [=client device=]" ?

File: index.bs [hunk 6]
```diff
@@ -918,14 +918,19 @@ The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "S
...
+    This is in contrast to a [=server-side-resident public key credential source=],
+    which requires that the [=authenticator=] is given both the [=RP ID=] and the [=credential ID=]
+    but does not require [=client-side=] storage of the [=credential private key=].
```
Comment (equalsJeffH) [inline index.bs]: should "[=credential private key=]" here be "[=public key credential source=]" ?

File: index.bs [hunk 7]
```diff
@@ -1076,9 +1081,17 @@ The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "S
 
 : <dfn>Server-side-resident Public Key Credential Source</dfn>
 : <dfn>Non-Resident Credential</dfn>
-:: A [=Server-side-resident Public Key Credential Source=], or [=Non-Resident Credential=] for short, is a [=public key credential
-    source=] whose [=credential private key=] is not stored in the [=authenticator=], [=client=] or [=client device=], but is instead
-    encrypted (i.e. wrapped) and returned to the server as the [=credential ID=]. See also: [=server-side credential storage modality=].
+:: A [=Server-side-resident Public Key Credential Source=], or [=Non-Resident Credential=] for short,
+    is a [=public key credential source=] that cannot be used in an [=authentication ceremony=]
+    without providing the [=authenticator=] with the [=credential ID=], e.g.,
+    it can only be used when specified in the {{PublicKeyCredentialRequestOptions/allowCredentials}} argument.
```
Comment (equalsJeffH) [inline index.bs]:
```suggestion
    the credential is only usable when its [=credential Id=] is specified in the {{PublicKeyCredentialRequestOptions/allowCredentials}} argument.
```

File: index.bs [hunk 8]
```diff
@@ -1076,9 +1081,17 @@ The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "S
...
+
+    As a consequence, [=client-side=] storage of the [=credential private key=] is not required for a [=non-resident credential=].
```
Comment (equalsJeffH) [inline index.bs]: should "[=credential private key=]" here be "[=public key credential source=]" ?

File: index.bs [hunk 9]
```diff
@@ -1076,9 +1081,17 @@ The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "S
...
+    This is in contrast to a [=client-side-resident public key credential source=],
+    which does not require the [=credential ID=] to be used
```
Comment (equalsJeffH) [inline index.bs]:
```suggestion
    which does not require the [=credential ID=] to be provided
```

File: index.bs (hunk 5 above)
Comment (emlun) [reply index.bs]: Yeah, I guess that should work.

File: index.bs (hunk 7 above)
Comment (emlun) [reply index.bs]: Fixed case of ID:
```suggestion
    the credential is only usable when its [=credential ID=] is specified in the {{PublicKeyCredentialRequestOptions/allowCredentials}} argument.
```

File: index.bs (hunk 4 above)
Comment (emlun) [reply index.bs]: Yep, fixed.

File: index.bs [hunk 10]
```diff
@@ -918,16 +918,26 @@ The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "S
...
-    [=resident credential=], the [=authenticator=] might offload storage of wrapped key material to the
-    [=client device=], but the [=client device=] is not expected to offload the key storage to remote entities (e.g., [=[WRP]=] Server).
+:: A [=Client-side-resident Public Key Credential Source=], or [=Resident Credential=] for short,
+    is a [=public key credential source=] that is discoverable and usable in [=authentication ceremonies=]
+    without providing [=credential ID=]s,
+    i.e., calling {{CredentialsContainer/get()|navigator.credentials.get()}}
+    with an [=list/is empty|empty=] {{PublicKeyCredentialRequestOptions/allowCredentials}} argument.
+
+    As a consequence, a [=resident credential capable=] [=authenticator=] can generate an [=assertion signature=]
+    for a [=resident credential=] given only an [=RP ID=],
```
Comment (equalsJeffH) [inline index.bs]: s/ for / using / ?

Comment (emlun) [reply index.bs]: Hm, I see what you mean but I somehow feel like that muddies what I want to express here. I can't quite pinpoint what it is, but I think it has to do with how the "credential" in the challenger's (RP's) point of view is something the challenger has, and the signature is something to match it. Like the "credential" (public key) being a lock and the signature being a key _for_ it (even if the "key" is technically created _using_ the credential source).

Comment (equalsJeffH) [reply index.bs]: Yeah, I spose you're looking at it sorta differently than I do.  it's NBD (no big deal).

**Pull request merged**